### PR TITLE
WIP: Ana/moonpay integration

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Add Moonpay as our fiat on-ramp. Didn't do much yet actually. Saw their staging system is quite strange since you need to go through the whole verification process like normal and even add a regular card 🤷 So didn't get to the end of it yet

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
> Is this still relevant? If so, what is blocking it? Is there anything you can do to help move it forward?
> 
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

Yes, still super relevant. I realized the env variable eas not being picked up 😅  But yeah, still asking for a real credit card even though 😆

File: src/components/common/TmBalance.vue [hunk 1]
```diff
@@ -144,6 +144,14 @@
               <i class="material-icons">send</i></button
             ><span>Send</span>
           </div>
+          <div v-if="developmentMode" class="icon-button-container">
```
Comment (faboweb) [inline src/components/common/TmBalance.vue:147]: change to experimental mode?

File: src/components/common/TmBalance.vue [hunk 2]
```diff
@@ -250,6 +270,7 @@ export default {
       rewards: [],
       selectedFiatCurrency: "USD",
       preferredCurrency: "",
+      developmentMode: config.development,
```
Comment (faboweb) [inline src/components/common/TmBalance.vue:273]: remove

File: src/components/common/TmBalance.vue (hunk 1 above)
Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:147]: oh, ok. It is easier to test like this though. But ok

Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:147]: 👀 ?? What is the difference between experimental and developmentMode currently? In `session.js` I see:

```
    developmentMode: config.development, // can't be set in browser
    experimentalMode: config.development, // development mode, can be set from browser
```

**Comment by Bitcoinera** [comment]
I just contacted the Moonpay team asking on behalf of the credit card for their staging version. Let's see what they answer.

Also, saw that our US users won't be able to purchase any of the coins from our current networks

**Comment by jbibla** [comment]
design note: buttons should not be so close together:
https://www.sketch.com/s/8ffba04e-5055-44dd-984c-c60a74f5103b/a/mlOkjV

design question: should we have both buttons?

Screen Shot 2020-05-04 at 8 01 26 AM

File: src/components/common/TmBalance.vue [hunk 3]
```diff
@@ -190,6 +201,18 @@
               ><span>Send</span>
             </div>
           </div>
+
+          <div
```
Comment (jbibla) [inline src/components/common/TmBalance.vue:205]: note: the fact that we need to duplicate this indicates that there is a need for a refactor. https://github.com/luniehq/lunie/issues/3942#issuecomment-623424830

File: src/components/common/TmBalance.vue [hunk 4]
```diff
@@ -243,6 +266,7 @@ export default {
       rewards: [],
       selectedFiatCurrency: "USD",
       preferredCurrency: "",
+      moonpayCoins: new Set(["ATOM", "LUNA", "SDT", "KAVA"]), // TODO: add warning for US custormers as they cannot purchase any of these coins
```
Comment (jbibla) [inline src/components/common/TmBalance.vue:269]: why do we need to do `new Set()`?

File: src/components/common/TmBalance.vue [hunk 5]
```diff
@@ -761,6 +796,10 @@ select option {
   color: var(--menu-bright);
 }
 
+.circle-buy-button {
+  background: #f67f70;
```
Comment (jbibla) [inline src/components/common/TmBalance.vue:800]: this should be a variable — and the hover color should match

File: src/components/common/TmBalance.vue [hunk 6]
```diff
@@ -730,6 +761,10 @@ select option {
   padding-left: 1rem;
 }
 
+.icon-button-container {
+  margin-right: 5px;
```
Comment (jbibla) [inline src/components/common/TmBalance.vue:765]: use `rem` not pixels

**Comment by Bitcoinera** [comment]
I added a minimal fiat currency selector. But we do need this (if we want our users to be able to buy in other fiat currencies apart from USD), I confirmed it:

File: src/components/common/TmBalance.vue (hunk 6 above)
Comment (Bitcoinera) [reply src/components/common/TmBalance.vue]: It is actually the same in the end of the day no?

image
(our variables.css)

I changed it anyways

File: src/components/common/TmBalance.vue (hunk 4 above)
Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:269]: To know about if they are in the US or not we'd need to ask all our users for their location. They'd get a popup like this:



Or were we getting this info already for GA? 🤔

Comment (Bitcoinera) [reply src/components/common/TmBalance.vue:269]: This is done now. Will ask @jrmoreau to confirm it is working

File: src/components/common/TmBalance.vue (hunk 6 above)
Comment (jbibla) [reply src/components/common/TmBalance.vue]: no. `5px` is `5px` ... `var(--px)` is `0.0625rem`.

**Comment by jbibla** [comment]
so you think this is not available?

Screen Shot 2020-05-06 at 1 18 19 PM

i like the fiat selector but ... why is this separate from the existing currency dropdown?

**Comment by Bitcoinera** [comment]
> so you think this screen would not be available in our moonpay integration?

Where is this coming from? 😮 I hadn't seen that before! 

Show me how to use the fire 🔥 

> i like the fiat selector but ... why is this separate from the existing currency dropdown?

Yeah, this is something I have also considered... It could be tied together and for the non-supported currencies on Moonpay we would open the modal. And only then... But yeah, users need to realize this (that they are coupled together). So, that is why I think this is a better UX

**Comment by jbibla** [comment]
there's a menu in moonpay
Screen Shot 2020-05-06 at 2 05 07 PM

**Comment by jbibla** [comment]
ana, i think we should remove the fiat selector for now. it is not needed for a first implementation. 

please also fix the unit tests and conflicts.

let's make sure we can merge this code without activating it yet? @faboweb ?

**Comment by Bitcoinera** [comment]
> ana, i think we should remove the fiat selector for now. it is not needed for a first implementation.
> 
> please also fix the unit tests and conflicts.
> 
> let's make sure we can merge this code without activating it yet? @faboweb ?

Well, I can remove it but it will only allow people to buy crypto in USD. When ironically people in the USA cannot buy any of our tokens... 🤷 but as you prefer

**Comment by jbibla** [comment]
this is not a priority right now. let's revisit when we knock off some more of our projects on priorities list.

**Comment by Bitcoinera** [comment]
> this is not a priority right now. let's revisit when we knock off some more of our projects on priorities list.

OK, let's close it for now then

**Pull request closed without merging**